### ISSUE TITLE
Sharing, unsharing and re-sharing is not working correctly

### ISSUE THREAD

**Issue by gschoenberger** [nextcloud/passman#349]
## Bug report
Unshared credentials cannot be shared again, moreover they are breaking password entries.

### Steps to reproduce
1. Unshare a credential -> this disappears correctly on the receiver side.
![image](https://user-images.githubusercontent.com/1648307/29028589-19ffe62a-7b85-11e7-8419-31615715be85.png)
![image](https://user-images.githubusercontent.com/1648307/29028605-24fd75d8-7b85-11e7-99c2-1f02dee99829.png)

2. Try to share the credential again, seems to work at first glance. If you want to leave the share dialog the "this will corrupt the credential" error message appears
![image](https://user-images.githubusercontent.com/1648307/29028621-2ff2cf6a-7b85-11e7-94f8-e4559393cbcd.png)
![image](https://user-images.githubusercontent.com/1648307/29028632-382c7640-7b85-11e7-90e5-df82e3329242.png)

3. On the receiver side the credential does not appear again, but suddenly an other entry is listed twice.
![image](https://user-images.githubusercontent.com/1648307/29028666-565d1c5a-7b85-11e7-864a-cd8fc2556acd.png)

4. On the share side you cannot press the "unshare" button again.
![image](https://user-images.githubusercontent.com/1648307/29028681-623bcfee-7b85-11e7-9e6e-821c689112c9.png)

5. If I try to share it again, the receiver is listed as pending, but sharing did not work.
![image](https://user-images.githubusercontent.com/1648307/29029173-12e3a5b4-7b87-11e7-81e9-77c81c6463c0.png)

### Expected behaviour
If I press the "unshare" button at a certain credential -> the entry disappears on the receiver side
If I press "share" again -> the receiver side receives a notification for the shared credential
On both sides I want updated share icons for the credential

### Actual behaviour
The unshare button deletes the credential on the receiver side -> that's correct
A next share try leads to an incomplete share state -> a corruption error appears on the sharer side and on the receiver side you get duplicate entries
On the sharer side neither sharing nor unsharing is working anymore.

### Configuration
**Operating system**: 
Ubuntu 16.04 on server and client side 

**Browser**:   
Tested with Chrome Version 60.0.3112.78 (Official Build) (64-bit) and Firefox 54.0 (64-bit)

**Passman version**:    
Version: 2.1.4

**Database:**
MySQL 5.7.19-0ubuntu0.16.04.1

**PHP version:**
PHP-FPM 7.0.18-0ubuntu0.16.04.1

**cloud server:** Nextcloud
Nextcloud 12

**Comment by brantje** [comment]
We always use `Alice` (owner) and `Bob` (share receiver), to make things clear for our self.
Will update the issue to reflect `Alice` and `Bob` as soon i get time.

**Comment by gschoenberger** [comment]
I have updated the issue with Alice and Bob!

**Comment by animalillo** [comment]
Many thanks!
I will try to look into this this weekend if I have time! (No promises tho)

**Comment by KB7777** [comment]
I got the same bug with AD auth:

**Operating system:** CentOS 7
**Browser:** Tested with Chrome Version 59.0.3071.115
**Passman version:** Version: 2.1.4
**Database:** 10.1.25-MariaDB
**PHP version:** 7.1.8 from remi repo
**Cloud server:** Nextcloud 12.0.1

**Comment by gschoenberger** [comment]
Any updates on this issue?

**Comment by animalillo** [comment]
Yes, we have tracked it down.

Seems nextcloud is refusing to save null for the shared key, which is the way we detect if it's shared or not.

I might try to fix it this weekend, I have booked some time for passman.

**Comment by gschoenberger** [comment]
Nice! Really looking forward to use passman, but sharing is essential for me. Maybe I can effort some credits to donate - I will try my best :+1:

**Comment by ScreamingDev** [comment]
Does this issue also cover that Bob is not seeing the shared password (even on first share attempt)?
I mean: Alice is using "Share with users and groups" to give Bob the password but it does not pop-up in Bobs list.

I'm asking because I only see the link tab open in the screens and I currently have that issue.

**Comment by huynhcongdanh** [comment]
@ScreamingDev I dont think this issue is related. Do you have a vault created under Bob account?
--------------------
@brantje / @animalillo  Is there any update on this issue or some information that could help to debug? Passman is really nice. Im trying to onboard this tool for my team and ran into this issue.

After reading comment about "refuse to save null for the shared key", I thought why not creating a trigger to update the shared_key to null, but it seems to be more than that :) Look like this involve the logic of encrypt/decrypt the cred with shared_key="somestring". When unshared, the shared_key set to "", the cred seem to be re-encrypted or do something with that empty string and a simple trigger would not help :)

Anyway, I'm not familiar with the code and may take me forever to understand the whole magic. It would be great if I could get some explanation or something to start with.

Thanks very much and great works.

**Comment by mkreckovic** [comment]
Anything new about this issue? This is a huge problem, because if you unshare something you stay with a corupted password database.

**Comment by mkreckovic** [comment]
Is this project still alive?
